File: PopulateDeleteNode.java
class PopulateDeleteNode{

    LinkedList firstNode;
    LinkedList traverser;

    public PopulateDeleteNode(){
    
        firstNode = null;
    
    }
    
    public void addNode(int data){
        
        
        LinkedList listVal = new LinkedList(data);

        traverser = firstNode;
        
        if(traverser == null){
        
            firstNode = listVal;           
        
        }

        else{

            while( traverser.next != null){
        
                traverser = traverser.next;
        
            }

            traverser.next = listVal;
        
        }

    }

    public void viewNodes(){
        
        traverser = firstNode;
        
        while(traverser!=null){
        
            System.out.print(traverser.getData()+" ");
            traverser = traverser.getNext(); 
        }
        
    }

    public LinkedList getHead(){
    
        
        return this.firstNode;
    
    
    }

    public void deleteNode(LinkedList node){
    
        traverser = firstNode;

        if(traverser == null){
        
            System.out.println("List is empty");
        
        }

        else if(traverser == node){
        
            firstNode = traverser.getNext();

            System.out.println("Node deleted with value: "+traverser.getData());
            traverser = null; 
        }

        else{

        
        
            LinkedList prev = firstNode;
            traverser = traverser.getNext();

            while(traverser != null){
            
                if(traverser == node){
                
                    prev.setNext(traverser.getNext());

                    System.out.println("Node deleted with value: "+traverser.getData());
                
                    traverser = null;
                    break;
                }
         
                traverser = traverser.getNext();
                prev = prev.getNext();
            }


        }
    
    }


    public void makeLinkedList(){
    
        
            addNode(1);
            addNode(3);
            addNode(4);
            addNode(0);
            addNode(10);
            addNode(6);
            addNode(4);
            addNode(1);
            addNode(8);
            addNode(9);
            addNode(8);
            addNode(8);
            viewNodes();
    
    }


}

File: PopulateDeleteNode.c
#include<stdio.h>
#include<stdlib.h>



typedef struct node{

    int data;
    struct node *next;


}linkedlist;





linkedlist *addToList(linkedlist *list,int value){
    
    linkedlist *traverser = list;
    
    linkedlist *node;
    node = (linkedlist *)malloc(sizeof(linkedlist));
    node->data = value;
    node->next = NULL;

    if(traverser == NULL){
        
        printf("Set first Node");
        list  = node;
    
    }
    
    else{

        while(traverser->next != NULL){
            
            traverser = traverser -> next;
            
        }

        traverser->next = node;
    }

    return list;


}



void printList(linkedlist *list){

    
    linkedlist *traverser = list;
    
    while(traverser != NULL){
    
        printf("\t %d",traverser->data);
   
        traverser = traverser -> next;

    }

    printf("\n");

}


void deletelist(linkedlist *list){

   printf("\nFreed list\n");
    
    
    linkedlist *traverser = list;

    linkedlist *temp;


    if(traverser == NULL){
    
        printf("\nList is Empty\n");
    
    }



    
    while(traverser != NULL){
    
        temp = traverser;
        traverser = traverser->next;
        free(temp);
    
    
    }


}

int main(){

    linkedlist * list = NULL;

    list = addToList(list,1);
    list = addToList(list,3);
    list = addToList(list,3);
    list = addToList(list,3);
    list = addToList(list,6);
    list = addToList(list,2);
    list = addToList(list,0);
    list = addToList(list,6);
   
    printList(list);

    deletelist(list);

    return 0;

}
